perllib/FixMyStreet/App/Controller/Reports.pm

```diff
@@ -438,13 +445,25 @@ sub ward_check : Private {
     my $qw = $c->cobrand->fetch_area_children($parent_id);
     $c->cobrand->call_hook('add_parish_wards' => $qw);
 
+    if ($c->stash->{ward_code}) {
+        for my $key (keys %$qw) {
+            if (${$qw->{$key}}{type} ne $c->stash->{ward_code}) {
+                delete $qw->{$key};
+            }
+        };
```

Comment:
Instead of this block, how about something like:
```
$qw = values %$qw;
if ($c->stash->{ward_code}) {
    @$qw = grep { $_->{type} ne $c->stash->{ward_code} } @$qw;
}
```
then can use @$qw below rather than `values %$qw`.